solutions/usaco-834.mdx

```diff
@@ -0,0 +1,68 @@
+---
+id: usaco-1037
+source: USACO Bronze 2020 US Open
+title: Cowntact Tracing
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_sort_silver_open18.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N \cdot T^2)$
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class OutOfSorts {
+
+	// hold values
+	static class Entry implements Comparable<Entry> {
+		public int value;
+		public int index;
+
+		public Entry (int v, int i) {
+			value = v;
+			index = i;
+		}
+
+		public int compareTo(Entry other) { // bubble sort
+			if (this.value != other.value) {
+				return this.value - other.value;
+			}
+			return this.index - other.index;
+		}
+	}
+
+	public static void main(String[] args) throws Exception {
+		Kattio io = new Kattio("sort");
+		int n = io.nextInt();
+		Entry[] entries = new Entry[n];
+		for (int i = 0; i < n; i++) {
+			entries[i] = new Entry(io.nextInt(), i);
+		}
+		Arrays.sort(entries); // perform bubble sort
```

Comment:
arrays.sort does not do bubble sort

Reply:
i think if the class Entry has a compareTo method, the sorting will use it

Reply:
the default sort method for objects in java is merge sort, not bubble sort

Reply:
yes, it doesn't do bubble sort. i realized that it doesn't actually perform bubble sort. the sorting just makes it easier to compute the number of moos